pkg/controller/infinispan/autoscale.go

```diff
@@ -0,0 +1,178 @@
+package infinispan
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	infinispanv1 "github.com/infinispan/infinispan-operator/pkg/apis/infinispan/v1"
+	"github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	consts "github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// Starting a go routine that does polling autoscaling
+func addAutoscalingEquipment(mgr manager.Manager, r reconcile.Reconciler) {
+	go autoscalerPoll(mgr, r)
+}
+
+var defaultMinimumPollPeriod = time.Second
+
+func autoscalerPoll(mgr manager.Manager, r reconcile.Reconciler) {
+	log.Info("Starting adhoc loop for autoscaling")
+	for true {
+		time.Sleep(defaultMinimumPollPeriod)
+		ispnList := infinispanv1.InfinispanList{}
+		// Check all the cluster in the namespace for autoscaling
+		err := mgr.GetClient().List(context.TODO(), &ispnList)
+		if err != nil {
+			log.Error(err, "Unable to select cluster pods list")
+			continue
+		}
+		for _, iItem := range ispnList.Items {
```

Comment:
@rigazilla looks like sequential processing is a little bit dangerous if one of the working step in multiple cluster env is hung. Wdyt to split every cluster processing as they are independent of each other in separate goroutines and interact with channels?

Reply:
I see your point. Yeah, maybe we can make things more robust.

Reply:
@dmvolod wdyt about a goroutine per cluster? We can use the recocile loop to start and stop the scale monitor goroutine during cluster creation/destruction

Reply:
Yes, @rigazilla it'a s good options. I collect some ideas, we have check on implementation
- verify that only one thread per cluster is working simultaneously
- avoid conflicts between auto- and manual scaling
- validate up-scaling/descaling rate to avoid it doing very frequently